### PR TITLE
Feature/125 backend validation

### LINKED ISSUE #125: Backend validation for create/update workflow endpoint
Backed validation for create/update workflow endpoint.
Schema definition for a business error message.

### PR DESCRIPTION
**Pull request by kevinwallimann**
Closes #125

### CODE REVIEW

**Comment by kevinwallimann** [comment]
Closed PR because the Update part is not done yet

File: src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/controllers/WorkflowController.scala [hunk 1]
```diff
@@ -17,20 +17,28 @@ package za.co.absa.hyperdrive.trigger.api.rest.controllers
...
 
-import scala.concurrent.ExecutionContext.Implicits.global
 import scala.compat.java8.FutureConverters._
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
 
 @RestController
 class WorkflowController @Inject()(workflowService: WorkflowService) {
 
+  implicit def eitherToCompletableFutureOrException[T](response: Future[Either[Seq[ApiError], T]]): CompletableFuture[T] =
```
Comment (jozefbakus) [inline src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/controllers/WorkflowController.scala:33]: I would move this to Controller interface/class and implement it here and we can extract .toJava.toCompletableFuture as well

File: src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/services/WorkflowService.scala [hunk 2]
```diff
@@ -39,10 +41,30 @@ trait WorkflowService {
...
+                          override val workflowValidationService: WorkflowValidationService) extends WorkflowService {
+
+  def createWorkflow(workflow: WorkflowJoined)(implicit ec: ExecutionContext): Future[Either[Seq[ApiError], Boolean]] = {
+    for {
+      validationErrors <- workflowValidationService.validateOnInsert(workflow)
+      dbError <- doIf(validationErrors.isEmpty, () => workflowRepository.insertWorkflow(workflow), None)
+    } yield { toEither(Seq(validationErrors, dbError.map(error => Seq(error)))) }
+  }
+
+  private def doIf[T](condition: Boolean, future: () => Future[T], defaultValue: T) = {
```
Comment (jozefbakus) [inline src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/services/WorkflowService.scala]: I would move it to the end or the beginning of the file.

**Comment by jozefbakus** [comment]
Looks good only those two minor things. I will approve once I have a response on my notes.

File: src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/controllers/WorkflowController.scala (hunk 1 above)
Comment (kevinwallimann) [reply src/main/scala/za/co/absa/hyperdrive/trigger/api/rest/controllers/WorkflowController.scala:33]: As discussed, won't do now

**Review by jozefbakus — APPROVED** [review]
👍

**Pull request merged**